cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesExport.java

```diff
@@ -0,0 +1,408 @@
+                LOG.info(log.get());
+                continue;
+            }
+            writer.writeStartElement(CIM_NAMESPACE, CgmesNames.TOPOLOGICAL_ISLAND);
+            writer.writeAttribute(RDF_NAMESPACE, ID, getUniqueId());
+            writer.writeStartElement(CIM_NAMESPACE, CgmesNames.NAME);
+            writer.writeCharacters(getUniqueId()); // TODO do we need another name?
+            writer.writeEndElement();
+            writer.writeEmptyElement(CIM_NAMESPACE, "TopologicalIsland.AngleRefTopologicalNode");
+            writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, "#" + angleRefs.get(island.getKey()));
+            for (String tn : island.getValue()) {
+                writer.writeEmptyElement(CIM_NAMESPACE, "TopologicalIsland.TopologicalNodes");
+                writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, "#" + tn);
+            }
+            writer.writeEndElement();
+        }
+    }
+
+    private static Map<String, String> buildAngleRefs(Network network) {
+        Map<String, String> angleRefs = new HashMap<>();
+        for (VoltageLevel vl : network.getVoltageLevels()) {
+            SlackTerminal slackTerminal = vl.getExtension(SlackTerminal.class);
+            if (slackTerminal != null && slackTerminal.getTerminal() != null) {
+                if (slackTerminal.getTerminal().getBusBreakerView().getBus() != null && slackTerminal.getTerminal().getBusBreakerView().getBus().getSynchronousComponent() != null) {
+                    String componentNum = String.valueOf(slackTerminal.getTerminal().getBusBreakerView().getBus().getSynchronousComponent().getNum());
+                    if (angleRefs.containsKey(componentNum)) {
+                        Supplier<String> log = () -> String.format("Several slack buses are defined for synchronous component %s: only first slack bus (%s) is taken into account",
+                                componentNum, angleRefs.get(componentNum));
+                        LOG.info(log.get());
+                        continue;
+                    }
+                    angleRefs.put(componentNum, slackTerminal.getTerminal().getBusBreakerView().getBus().getId());
+                } else if (slackTerminal.getTerminal().getBusBreakerView().getBus() != null) {
+                    angleRefs.put(slackTerminal.getTerminal().getBusBreakerView().getBus().getId(),
+                            slackTerminal.getTerminal().getBusBreakerView().getBus().getId());
+                } else {
+                    Supplier<String> message = () -> String.format("Slack terminal at equipment %s is not connected and is not exported as slack terminal", slackTerminal.getTerminal().getConnectable().getId());
+                    LOG.info(message.get());
+                }
+            }
+        }
+        return angleRefs;
+    }
+
+    private static Map<String, List<String>> buildIslands(Network network) {
```

Comment:
Maybe put a comment to say that it works only for bus branch topology.

Reply:
Same remark